docs/file-layout.md

```diff
@@ -0,0 +1,98 @@
+---
+id: file-layout
+title: File Layout
+---
+
+MUnit provides some utilities to test programs that manipulate files.
+
+You can use `FileLayout` to describe the content of files on disk and either
+write them to use them as input for your test or check that the files on disk
+are as expected.
+
+Example:
+
+suppose you are testing a CLI tool that can read credentials from a file
+
+```scala mdoc
+import java.nio.file.Paths
+import java.nio.file.Files
+
+object MyCliTool {
+  def main(args: Array[String]): Unit =
+    credentialsFromFile(args(0)) match {
+      case Some(value) => println("Credentials found!")
+      case None        => println("No credentials found :(")
+    }
+
+  def credentialsFromFile(configPath: String): Option[String] = {
+    val path = Paths.get(configPath)
+    if (Files.exists(path))
+      Some(new String(Files.readAllBytes(Paths.get(configPath))))
+    else
+      None
+  }
+}
+```
+
+You can `FileLayout` to write a test for reading the credentials:
+
+```scala mdoc
+import munit.FunSuite
+import munit.FileLayout
+
+class MyCliSuite extends FunSuite {
+
+  test("credentials-from-file") {
+    val authPath = ".myCli/auth"
+    val expectedToken = "token=abcdef123456"
+    val root = FileLayout.write(
+      s"""|/${authPath}
```

Comment:
I feel like we can come up with a better motivating example. In this specific example we're not taking much advantage of the multiple-line syntax, it might even be clearer if anything with a `Map[String, String]` data structure.

The biggest value in file layouts is when you have large literal strings and minimal interpolated variables, ideally with multiple files in nested directories (like the `D.scala` and `e.json` example below).